test/main.go

```diff
@@ -13,7 +13,9 @@ const templateFile = "../dir-index.html"
 // Copied from go-ipfs/core/corehttp/gateway_indexPage.go
```

Comment:
I believe the updated `dir-index.html` template can be safely merged and be used upstream, after the type declarations below have been included. All of the new struct fields are optional and the updated `dir-index.html` template can be safely used without them. However, to see most of the UI improvements provided by of this PR, the new struct fields will need to be populated as in the code below.

Reply:
@jessicaschilling @neatonk it looks like the changes in this library are fine, but we can't reasonably merge this until the corresponding changes are made in the go-ipfs gateway to emit the new template struct fields.

@jessicaschilling Is implementing the changes to go-ipfs part of the bounty as well?

Reply:
It wasn’t in the original bounty spec - however, @neatonk, is that something you could take on? We may be able to adjust the bounty as a result.

Reply:
Yep, I can do that. 

Regarding the bounty, I think it would be best to either increase the payment to reflect the additional work needed or close it out for the work that was done. I am fine with either option.